### PR TITLE
Support rack 3

### PR DESCRIPTION
**Pull request by sl0thentr0py**
* Fix `is_server_protocol?` check for rack 3 support
* fix rack lint spec since uppercase no longer allowed in response headers
 * expand CI matrix to test on rack 2 & 3
 * remove `bundler: 1` pin in actions
* ~sqlite3 1.5.0 has pre-built binaries from ruby 2.6 onwards so use `force_ruby_platform` for 2.4 and 2.5 for `delayed_job`~

### CODE REVIEW

File: sentry-ruby/lib/sentry/interfaces/request.rb
Comment (antonpirker) [inline sentry-ruby/lib/sentry/interfaces/request.rb:132]: This will be wrong in Rack 4, no?

Comment (sl0thentr0py) [reply sentry-ruby/lib/sentry/interfaces/request.rb:132]: Rack has a major bump once in 5 years or so, think of it like WSGI, it doesn't change much.

Comment (sl0thentr0py) [reply sentry-ruby/lib/sentry/interfaces/request.rb:132]: But yea, i guess I could do an int here, thx!

**Comment by sl0thentr0py** [comment]
ughh now sqlite is failing 😭

File: sentry-ruby/Gemfile
Comment (st0012) [inline sentry-ruby/Gemfile:6]: Do you think we can use `RACK_VERSION=0` for this?

File: .github/workflows/sentry_ruby_test.yml
Comment (st0012) [inline .github/workflows/sentry_ruby_test.yml:24]: We can probably get this and the one below by adding `3.1` to the `ruby_version` array?

File: .github/workflows/sentry_delayed_job_test.yml
Comment (st0012) [inline .github/workflows/sentry_delayed_job_test.yml:47]: TIL this config. What does it do?

Comment (sl0thentr0py) [reply .github/workflows/sentry_delayed_job_test.yml:47]: The latest [sqlite3 release](https://github.com/sparklemotion/sqlite3-ruby/releases/tag/v1.5.0) only publishes binaries from ruby 2.6 onwards and CI was failing on 2.4/2.5 because bundler was trying to look for binaries instead of installing the ruby platform version (compiling native extensions). 

This config forces bundler to always compile on 2.4 and 2.5.

File: sentry-ruby/Gemfile
Comment (sl0thentr0py) [reply sentry-ruby/Gemfile:6]: yes we can simplify!

File: .github/workflows/sentry_ruby_test.yml
Comment (sl0thentr0py) [reply .github/workflows/sentry_ruby_test.yml:24]: the matrix was doing some weird stuff with includes, so I just left all the 3.1 ones explicitly

**Comment by sl0thentr0py** [comment]
@st0012 also one more thing I noticed, we have [`bundler: 1`](https://github.com/getsentry/sentry-ruby/blob/4304412dab8c09cfd3ce1e1914199311003f0659/.github/workflows/sentry_ruby_test.yml#L34) in all the github actions, any reason we're pinning to such an old bundler version?

**Comment by st0012** [comment]
@sl0thentr0py I suspect it may be related to Ruby 2.4 support? We can try removing that config though.

File: .github/workflows/sentry_ruby_test.yml
Comment (sl0thentr0py) [reply .github/workflows/sentry_ruby_test.yml]: yea, it just runs these with the simplified matrix, no clue why
image

File: .github/workflows/sentry_delayed_job_test.yml
Comment (sl0thentr0py) [reply .github/workflows/sentry_delayed_job_test.yml]: ok this is no longer needed with latest bundler, removed

File: sentry-ruby/lib/sentry/interfaces/request.rb
Comment (st0012) [inline sentry-ruby/lib/sentry/interfaces/request.rb:133]: @sl0thentr0py consider we have this change, it probably still needs a changelog entry.

**Review by st0012 — COMMENTED** [review]
All look good, just have a small comment. Great job 🙌

**Comment by sl0thentr0py** [comment]
@st0012 can you approve if ok? (we added a required check for approval now for compliance reasons, so need it to merge)

**Review by st0012 — APPROVED** [review]

**Pull request merged**
